File: 168.excel-sheet-column-title.java
/*
 * @lc app=leetcode id=168 lang=java
 *
 * [168] Excel Sheet Column Title
 */

// @lc code=start
class Solution {
    public String convertToTitle(int columnNumber) {
        
        StringBuilder res = new StringBuilder();
        
        while (columnNumber > 0) {
            columnNumber--;
            res.append((char)(columnNumber % 26 + 'A'));
            columnNumber = columnNumber / 26;
        }

        return res.reverse().toString();
    }
}
// @lc code=end

File: 168.excel-sheet-column-title.py
#
# @lc app=leetcode id=168 lang=python
#
# [168] Excel Sheet Column Title
#

# @lc code=start
class Solution(object):
    def convertToTitle(self, columnNumber):
        """
        :type columnNumber: int
        :rtype: str
        """
        left = columnNumber
        res = ""

        while left > 0:
            left, reminder = divmod(left - 1, 26)
            res = chr(65 + reminder) + res
        return res
                 
        
# @lc code=end

